WTWSMS/events/WTWSMS_historical_events.txt

```diff
@@ -1127,3 +1127,30 @@ narrative_event = {
 		prestige = 50 
 	}
 }
+
+# Extreme weather events of 535-536
+narrative_event = {
+	id = hist.26
+	title = hist.26.name
+	desc = hist.26.desc
+	picture = GFX_evt_autumn_harvest
+	border = GFX_event_narrative_frame_diplomacy
+	
+	only_playable = yes
+	major = yes
+	
+	hide_from = yes
+
+	is_triggered_only = yes #on_action
+	
+	trigger = {
+		year = 536
+		NOT = { has_global_flag = 535_536_weather_events }
+	}
```

Comment:
For who do you want this event to fire? You set it as `major = yes` but you don't have any recipient. Usually you would either set a ruler that always stays, select a particular character or have it fire individually.

Reply:
For one random person, then the rest receives it because of `major = yes`.

Reply:
Then set it to a fix ruler who always exists.

Reply:
What would it improve? The only difference I can see is more complex trigger, which leads to lower efficiency.

Reply:
It would ensure the event is always displayed properly as all other events behave in the mod.

Reply:
But even with your solution it evaluates for all rulers, just with one condition more.

Reply:
That logic is based upon a different understanding of events calculation than the one I have. Because the random character you speak of might be two, or three, since the trigger is evaluated separately. Then they would get three sets of events each, which is broken.

Reply:
I have modified the event to only trigger for the holder of a certain province (same as in the maintenence event), it should do the job.

Reply:
Ok